src/v/archival/ntp_archiver_service.cc

```diff
@@ -595,59 +651,75 @@ ntp_archiver::schedule_uploads(model::offset last_stable_offset) {
                                                      + model::offset(1)
                                                  : model::offset(0);
 
-    vlog(
-      _rtclog.debug,
-      "scheduling uploads, start_upload_offset: {}, last_stable_offset: {}",
-      start_upload_offset,
-      last_stable_offset);
-    _probe.upload_lag(last_stable_offset - start_upload_offset);
+    auto compacted_segments_upload_start = model::next_offset(
+      manifest().get_last_uploaded_compacted_offset());
+
+    std::vector<upload_context> params;
+    if (_partition->get_ntp_config().is_compacted()) {
```

Comment:
I think that non-compacted uploads should go first and have a priority over compacted. If the upload of non-compacted segments will be lagging the compaction will be delayed.

Reply:
moved non-compacted to upload first